### ISSUE TITLE
FeatureDev locks up IDE while building context on a large project

### ISSUE THREAD

**Issue by rli** [aws/aws-toolkit-jetbrains#4269]
```
- SupervisorJobImpl{Active}
	- "software.aws.toolkits.jetbrains.services.amazonqFeatureDev.FeatureDevApp":StandaloneCoroutine{Active}, state: SUSPENDED [com.intellij.util.concurrency.AppScheduledExecutorService$BackendThreadPoolExecutor@3da770a1[Running, pool size = 7, active threads = 4, queued tasks = 0, completed tasks = 321285]]
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.FeatureDevApp$init$1.invokeSuspend(FeatureDevApp.kt:47)
	- "software.aws.toolkits.jetbrains.services.amazonqFeatureDev.FeatureDevApp":StandaloneCoroutine{Active}, state: RUNNING [com.intellij.util.concurrency.AppScheduledExecutorService$BackendThreadPoolExecutor@3da770a1[Running, pool size = 7, active threads = 4, queued tasks = 0, completed tasks = 321285]]
		at java.base@17.0.10/java.lang.Character.codePointAt(Character.java:8910)
		at java.base@17.0.10/java.util.regex.Pattern$CharPropertyGreedy.match(Pattern.java:4273)
		at java.base@17.0.10/java.util.regex.Pattern$CharPropertyGreedy.match(Pattern.java:4290)
		at java.base@17.0.10/java.util.regex.Pattern$Start.match(Pattern.java:3608)
		at java.base@17.0.10/java.util.regex.Matcher.search(Matcher.java:1728)
		at java.base@17.0.10/java.util.regex.Matcher.find(Matcher.java:745)
		at kotlin.text.Regex.containsMatchIn(Regex.kt:109)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.session.FeatureDevSessionContext.ignoreFile(FeatureDevSessionContext.kt:55)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.session.FeatureDevSessionContext.access$ignoreFile(FeatureDevSessionContext.kt:21)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.session.FeatureDevSessionContext$zipFiles$1.invoke(FeatureDevSessionContext.kt:60)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.session.FeatureDevSessionContext$zipFiles$1.invoke(FeatureDevSessionContext.kt:58)
		at software.aws.toolkits.core.utils.ZipUtilsKt.createTemporaryZipFile(ZipUtils.kt:40)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.session.FeatureDevSessionContext.zipFiles(FeatureDevSessionContext.kt:58)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.session.FeatureDevSessionContext.access$zipFiles(FeatureDevSessionContext.kt:21)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.session.FeatureDevSessionContext$getProjectZip$zippedProject$1.invoke(FeatureDevSessionContext.kt:48)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.session.FeatureDevSessionContext$getProjectZip$zippedProject$1.invoke(FeatureDevSessionContext.kt:48)
		at com.intellij.openapi.application.ActionsKt.runReadAction$lambda$3(actions.kt:31)
		at com.intellij.openapi.application.ActionsKt$$Lambda$1823/0x00000008011d8bf8.compute(Unknown Source)
		at com.intellij.openapi.application.impl.RwLockHolder.runReadAction(RwLockHolder.kt:271)
		at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:845)
		at com.intellij.openapi.application.ActionsKt.runReadAction(actions.kt:31)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.session.FeatureDevSessionContext.getProjectZip(FeatureDevSessionContext.kt:48)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.session.PrepareRefinementState.interact(PrepareRefinementState.kt:16)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.session.Session.nextInteraction(Session.kt:145)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.session.Session.send(Session.kt:137)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.controller.FeatureDevController.onApproachGeneration(FeatureDevController.kt:485)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.controller.FeatureDevController.handleChat(FeatureDevController.kt:400)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.controller.FeatureDevController.processPromptChatMessage(FeatureDevController.kt:79)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.FeatureDevApp.handleMessage(FeatureDevApp.kt:73)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.FeatureDevApp.access$handleMessage(FeatureDevApp.kt:21)
		at software.aws.toolkits.jetbrains.services.amazonqFeatureDev.FeatureDevApp$init$1$1$1.invokeSuspend(FeatureDevApp.kt:49)

- SupervisorJobImpl{Active}
	- "software.aws.toolkits.jetbrains.services.codemodernizer.CodeTransformChatApp":StandaloneCoroutine{Active}, state: SUSPENDED [com.intellij.util.concurrency.AppScheduledExecutorService$BackendThreadPoolExecutor@3da770a1[Running, pool size = 7, active threads = 4, queued tasks = 0, completed tasks = 321285]]
		at software.aws.toolkits.jetbrains.services.codemodernizer.CodeTransformChatApp$init$1.invokeSuspend(CodeTransformChatApp.kt:69)
		- "software.aws.toolkits.jetbrains.services.codemodernizer.CodeTransformChatApp":ProducerCoroutine{Completing} [com.intellij.util.concurrency.AppScheduledExecutorService$BackendThreadPoolExecutor@3da770a1[Running, pool size = 7, active threads = 4, queued tasks = 0, completed tasks = 321285]]
			-[x2 of] "software.aws.toolkits.jetbrains.services.codemodernizer.CodeTransformChatApp":StandaloneCoroutine{Active}, state: SUSPENDED [com.intellij.util.concurrency.AppScheduledExecutorService$BackendThreadPoolExecutor@3da770a1[Running, pool size = 7, active threads = 4, queued tasks = 0, completed tasks = 321285]]
				at kotlinx.coroutines.flow.internal.ChannelLimitedFlowMerge$collectTo$2$1.invokeSuspend(Merge.kt:96)
```

**Comment by rli** [comment]
```
$ du -h -d1
9.1G    .
```

**Comment by andrewyuq** [comment]
The UI freeze issue is fixed in latest Amazon Q version 3.1
